rules_jvm_export/jvm_export/jvm_export.bzl

```diff
@@ -0,0 +1,385 @@
+# Based on https://github.com/vaticle/bazel-distribution/blob/master/maven/rules.bzl
+# SPDX-License-Identifier: Apache-2.0
+
+JvmExportInfo = provider(
+    fields = {
+        "jar": "JAR file to deploy",
+        "srcjar": "JAR file with sources",
+        "pom": "Accompanying pom.xml file"
+    }
+)
+
+
+VersionInfo = provider(
+    doc = "A singleton provider that contains the raw value of a build setting",
+    fields = {
+        "value": "Version number"
+    },
+)
+
+
+def _jvm_export_impl(ctx):
+    version = ctx.attr.version[VersionInfo].value
+    pom_file = _generate_pom_file(ctx, version)
+    if ctx.attr.assemble:
+      class_jar = _generate_class_jar(ctx, pom_file)
+    else:
+      class_jar = ctx.attr.target[JavaInfo].runtime_output_jars[0]
```

Comment:
I guess we don't know for sure, but mostly we need to support rules_scala and `native.java_library` so if it works for those we should be ok?

Reply:
Added

```python
def _class_jar(target):
    if len(target[JavaInfo].runtime_output_jars) == 1:
        return target[JavaInfo].runtime_output_jars[0]
    else:
        fail(
            "expected size 1, but runtime_output_jars in {} was {}".format(
                target, target[JavaInfo].runtime_output_jars
            )
        )
```

so at least this will fail the `run` if there's more than one JARs associated with the output.